### PR TITLE
Fix a bug in inline assembly llvm IR generation

### PR DESCRIPTION
**Pull request by woehr**
This fixes a bug in LLVM IR generation for in-line assembly where Rust would always use the host clobbers instead of target clobbers.

I also took this opportunity to clean/simplify the `trans_inline_asm` function.

This is my first Rust pull request; please let me know if I'm missing anything.

### CODE REVIEW

**Comment by richo** [comment]
Great first pr!!

(Haven't read diff properly and can't r+, just wanted to thank and welcome)

File: src/librustc_trans/trans/asm.rs [hunk 1]
```diff
@@ -76,43 +76,34 @@ pub fn trans_inline_asm<'blk, 'tcx>(bcx: Block<'blk, 'tcx>, ia: &ast::InlineAsm)
...
-    // Add the clobbers to our constraints list
-    if clobbers.len() != 0 && constraints.len() != 0 {
-        constraints.push(',');
-        constraints.push_str(&clobbers[..]);
-    } else {
-        constraints.push_str(&clobbers[..]);
-    }
+    let clobbers = ia.clobbers.iter()
+                              .map(|s| format!("~{{{}}}", &s))
+                              .collect::<Vec<String>>();
```
Comment (huonw) [inline src/librustc_trans/trans/asm.rs]: I suspect this `collect` may be unnecessary, and `.chain(clobbers.into_iter())` below can become `.chain(clobbers)`.

**Comment by woehr** [comment]
I addressed the unnecessary vector creation that @huonw pointed out. In doing so I now realize I accidentally changed an ```into_iter``` to an ```iter``` so I'd like to ask, is there a preference to which is used over vectors in this case?

**Comment by alexcrichton** [comment]
> is there a preference to which is used over vectors in this case?

It largely depends on the use case at hand (whether you want to transfer ownership or not) so there's not really a hard-and-fast convention on which to prefer. For small cases like this it's generally whichever works out best!

@bors: r+ b92fee9

Thanks!

**Pull request merged**